contracts/crosschain/wormhole/WormholeGatewayAdapter.sol

```diff
@@ -0,0 +1,296 @@
+
+    /// @dev Returns the address of the remote gateway for a given EVM chain id.
+    function getRemoteGateway(uint256 chainId) public view virtual returns (address) {
+        address addr = _remoteGateways[chainId];
+        require(addr != address(0), UnsupportedChainId(chainId));
+        return addr;
+    }
+
+    /// @dev Registers a chain equivalence between a binary interoperable chain id and a Wormhole chain id.
+    function registerChainEquivalence(
+        bytes calldata chain,
+        uint16 wormholeId
+    ) public virtual /*onlyOwner in registerChainEquivalence*/ {
+        (uint256 chainId, ) = chain.parseEvmV1Calldata();
+        registerChainEquivalence(chainId, wormholeId);
+    }
+
+    /// @dev Registers a chain equivalence between an EVM chain id and a Wormhole chain id.
+    function registerChainEquivalence(uint256 chainId, uint16 wormholeId) public virtual onlyOwner {
+        require(
+            _chainIdToWormhole[chainId] == 0 && _wormholeToChainId[wormholeId] == 0,
+            ChainEquivalenceAlreadyRegistered(chainId, wormholeId)
+        );
+
+        _chainIdToWormhole[chainId] = wormholeId | MASK;
+        _wormholeToChainId[wormholeId] = chainId;
+        emit RegisteredChainEquivalence(chainId, wormholeId);
+    }
+
+    /// @dev Registers the address of a remote gateway (binary interoperable address version).
+    function registerRemoteGateway(bytes calldata remote) public virtual /*onlyOwner in registerRemoteGateway*/ {
+        (uint256 chainId, address addr) = remote.parseEvmV1Calldata();
+        registerRemoteGateway(chainId, addr);
+    }
+
+    /// @dev Registers the address of a remote gateway (EVM version).
+    function registerRemoteGateway(uint256 chainId, address addr) public virtual onlyOwner {
+        require(supportedChain(chainId), UnsupportedChainId(chainId));
+        require(_remoteGateways[chainId] == address(0), RemoteGatewayAlreadyRegistered(chainId));
+        _remoteGateways[chainId] = addr;
+        emit RegisteredRemoteGateway(chainId, addr);
+    }
+
+    /// @inheritdoc IERC7786GatewaySource
+    function supportsAttribute(bytes4 selector) public pure returns (bool) {
+        return selector == IERC7786Attributes.requestRelay.selector;
+    }
+
+    /// @inheritdoc IERC7786GatewaySource
+    function sendMessage(
+        bytes calldata recipient, // Binary Interoperable Address
+        bytes calldata payload,
+        bytes[] calldata attributes
+    ) external payable returns (bytes32 sendId) {
+        bool relay = false;
+        uint256 value = msg.value;
+        uint256 gasLimit = 0;
+
+        for (uint256 i = 0; i < attributes.length; ++i) {
+            (relay, value, gasLimit, ) = ERC7786Attributes.tryDecodeRequestRelayCalldata(attributes[i]);
+            require(relay, UnsupportedAttribute(attributes[i].length < 0x04 ? bytes4(0) : bytes4(attributes[i])));
+        }
+
+        if (relay) {
+            // TODO: Do we care about the returned "sequence"?
```

Comment:
Can we use the returned sequence instead of a custom `sendId`?

Reply:
Ah, we can't because we're using `sendId` for internal sequencing in the payload.

My point here was that we could return the sequence as the `sendId` from this function.